gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -95,6 +99,8 @@ public class TrinoQueryProperties
     private Set<String> catalogs = ImmutableSet.of();
     private Set<String> schemas = ImmutableSet.of();
     private Set<String> catalogSchemas = ImmutableSet.of();
+    private Optional<Call> procedure = Optional.empty();
+    private List<CallArgument> procedureArguments = ImmutableList.of();
```

Comment:
These fields are different from other values. Please move it under `isQueryParsingSuccessful` and leave a code comment explaining they are just for internal processing.